go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1239,6 +1239,45 @@ func (e *Executor) readPendingMigrationsUUIDs(ctx context.Context) (uuids []stri
 	return uuids, err
 }
 
+// readTablePendingMigrationsUUIDs returns UUIDs for migrations on given table, that are in pending state (queued/ready/running)
+func (e *Executor) readTablePendingMigrationsUUIDs(ctx context.Context, table string) (uuids []string, err error) {
+	r, err := e.execQuery(ctx, sqlSelectPendingMigrations)
+	if err != nil {
+		return uuids, err
+	}
+	// we identify running migrations on requested table
+	for _, row := range r.Named().Rows {
+		pendingUUID := row["migration_uuid"].ToString()
+		pendingKeyspace := row["keyspace"].ToString()
+		pendingTable := row["mysql_table"].ToString()
+
+		if pendingKeyspace == e.keyspace && pendingTable == table {
+			uuids = append(uuids, pendingUUID)
+		}
+	}
+	return uuids, err
+}
+
+// getLastCompletedMigrationOnTable returns the UUID of the last successful migration to complete on given table, if any
+func (e *Executor) getLastCompletedMigrationOnTable(ctx context.Context, table string) (found bool, uuid string, err error) {
```

Comment:
Nitpick: function name is "last completed", but docstring says "last successful", which don't necessarily mean the same thing, depending on your terminology.

EDIT: Looks like your success status is named "complete". So I guess this function name is in line with the terminology you've already chosen. But I think the fact that you used the wording "last successful" in the docstring betrays the fact that "complete" might not be clear enough :P